workers/social/lib/social/models/invitation.coffee

```diff
@@ -98,6 +100,26 @@ module.exports = class JInvitation extends jraphical.Module
       role          :
         type        : String
         default     : -> 'member'
+      inviter_id    :
+        type        : String
+
+  @revokeInvitation: permit 'remove invitation',
+    success: (client, invite, callback) ->
+
+      { connection: { delegate:account } } = client
+      account.fetchMyPermissions client, (err, permissionAndRoles) ->
+
+        { _id } = account
+        { roles } = permissionAndRoles
+
+        hasPermisson = 'admin' in roles or 'owner' in roles
+        { inviter_id, email, groupName } = invite
+
+        return callback 'noPermission'  if not hasPermisson and _id.toString() isnt inviter_id
```

Comment:
we use `KodingError` wrapper for such errors in the backend please check similar places for an example. cc/ @cihangir